src/cart/crt_context.c

```diff
@@ -806,19 +806,22 @@ crt_context_timeout_check(struct crt_context *crt_ctx)
 		crt_req_timeout_untrack(rpc_priv);
 
 		d_list_add_tail(&rpc_priv->crp_tmp_link, &timeout_list);
-		RPC_ERROR(rpc_priv,
-			  "ctx_id %d, (status: %#x) timed out, tgt rank %d, tag %d\n",
-			  crt_ctx->cc_idx,
-			  rpc_priv->crp_state,
-			  rpc_priv->crp_pub.cr_ep.ep_rank,
-			  rpc_priv->crp_pub.cr_ep.ep_tag);
 	};
 	D_MUTEX_UNLOCK(&crt_ctx->cc_mutex);
 
 	/* handle the timeout RPCs */
 	while ((rpc_priv = d_list_pop_entry(&timeout_list,
 					    struct crt_rpc_priv,
 					    crp_tmp_link))) {
+		RPC_ERROR(rpc_priv,
+			  "ctx_id %d, (status: %#x) timed out (%d seconds), "
+			  "target (%d:%d)\n",
+			  crt_ctx->cc_idx,
+			  rpc_priv->crp_state,
+			  rpc_priv->crp_timeout_sec,
```

Comment:
Are you sure that crp_timeout_sec is populated on all the code paths?

Reply:
i searched through code , it was not set in 1 case, will add a change in next upload to fix that.
In general:
rpc_priv->crp_timeout_sec is set to context timeout when rpc_priv is allocated
rpc_priv->crp_timeout_sec is also set when manually crt_req_set_timeout() is called on this rpc (after rpc_priv has been allocated).
The only place where it was missing is if neither of those are set, the default value was being used which was not reflected in rpc_priv->crp_timeout_sec